### PR TITLE
[AAP-77215] Add CleanTextMixin to DAB concrete model serializers

### PR DESCRIPTION
**Pull request by vidyanambiar**
#### ⚠️  Dependencies
This PR builds on top of https://github.com/ansible/django-ansible-base/pull/1087, which introduced the CleanTextMixin, Tier 1 name validator, and
Tier 2 free-text validator. That PR must be merged first — the commits from it appear in this branch's history.

  **What is being changed?**

See https://github.com/ansible/django-ansible-base/pull/1099/changes/8a0bbb527a99d2714f4a4db01d1293ca862d9b0e

  Wire `CleanTextMixin` (https://github.com/ansible/django-ansible-base/pull/1087) into DAB's five concrete model serializers so that Authenticator, AuthenticatorMap, OAuth2Application, OAuth2AccessToken, and
  RoleDefinition endpoints reject unsafe text input with HTTP 400.

  - `AuthenticatorSerializer` — Tier 1 validation on `name`
  - `AuthenticatorMapSerializer` — Tier 1 on `name`, with `excluded_fields` for `organization`/`role`/`team` (template expansion syntax)
  - `OAuth2ApplicationSerializer` — Tier 1 on `name`, Tier 2 on `description`
  - `OAuth2TokenSerializer` — Tier 2 on `description`
  - `RoleDefinitionSerializer` — Tier 1 on `name`, Tier 2 on `description`

  **Why is this change needed?**

  Addresses [AAP-77215](https://redhat.atlassian.net/browse/AAP-77215) under the parent epic [AAP-74584](https://redhat.atlassian.net/browse/AAP-74584) (DAB
  Input Validation and Sanitization). CAP-1040 identified that free-text API fields accept unsafe input (XSS, shell injection, control characters). Validation is
  enforced at the serializer layer so all clients (UI, API, CLI) get consistent rejection.

  **How does this change address the issue?**

  - `CleanTextMixin` is placed first in the MRO of each serializer, hooking into `validate()` via `super()` chaining
  - Two serializers (`AuthenticatorSerializer`, `AuthenticatorMapSerializer`) had broken `super().validate()` chains (`return data` instead of `return
  super().validate(data)`) — fixed as a prerequisite
  - `AuthenticatorMapSerializer` sets `excluded_fields = frozenset({'organization', 'role', 'team'})` because these fields accept `{% for_attr_value() %}`
  template expansion syntax
  - Subclass serializers (`AuthenticatorUpdateSerializer`, `RoleDefinitionDetailSerializer`) inherit the mixin automatically
  - Grandfathering: unchanged values on update are skipped, so pre-existing data is not blocked

  ## Type of Change
  - [ ] Bug fix (non-breaking change which fixes an issue)
  - [x] New feature (non-breaking change which adds functionality)
  - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - [ ] Documentation update
  - [x] Test update
  - [ ] Refactoring (no functional changes)
  - [ ] Development environment change
  - [ ] Configuration change

  ## Self-Review Checklist
  - [ ] I have performed a self-review of my code
  - [x] I have added relevant comments to complex code sections
  - [x] I have updated documentation where needed
  - [x] I have considered the security impact of these changes
  - [x] I have considered performance implications
  - [x] I have thought about error handling and edge cases
  - [ ] I have tested the changes in my local environment

  ## Testing Instructions

  ### Prerequisites
  - A working DAB test environment (`pip install -e ".[all]"`)

  ### Steps to Test

  1. Run the existing mixin unit tests:
     pytest test_app/tests/lib/serializers/test_clean_text_mixin.py -v
     pytest test_app/tests/lib/utils/test_validation.py -v

  2. Run the new integration tests for each app:
     pytest test_app/tests/authentication/test_clean_text_integration.py -v
     pytest test_app/tests/oauth2_provider/test_clean_text_integration.py -v
     pytest test_app/tests/rbac/api/test_clean_text_integration.py -v

  3. Run the existing serializer/view tests to verify no regressions:
     pytest test_app/tests/authentication/ -v
     pytest test_app/tests/oauth2_provider/ -v
     pytest test_app/tests/rbac/ -v

  4. Verify AuthenticatorMap template expansion fields still work:
     pytest test_app/tests/authentication/serializers/test_authenticator_map.py::TestAuthenticatorMapEscapeSequence -v

  5. Follow Testing Instructions here by setting the `sources.yaml` as described: https://docs.google.com/document/d/19vNXrfC-bnspJKWErvf0jgGStRqTYGmAsu54wWuuXwU/edit?usp=sharing

  ### Expected Results
  - All tests pass
  - Invalid names (e.g. `<script>alert(1)</script>`) rejected with HTTP 400 on create
  - Invalid descriptions (e.g. `$(rm -rf /)`) rejected with HTTP 400 on create
  - Pre-existing invalid values are grandfathered on update when unchanged
  - AuthenticatorMap template expansion syntax (`{% for_attr_value() %}`) still accepted in organization/role/team fields

  ## Additional Context

  ### Dependencies

  This PR builds on top of [#1087](https://github.com/ansible/django-ansible-base/pull/1087), which introduced the `CleanTextMixin`, Tier 1 name validator, and
  Tier 2 free-text validator. That PR must be merged first — the commits from it appear in this branch's history.

  ### Required Actions
  - [ ] Requires documentation updates
  - [x] Requires downstream repository changes
    
  - [ ] Requires infrastructure/deployment changes
  - [x] Requires coordination with other teams
    
  - [x] Blocked by PR/MR: [#1087](https://github.com/ansible/django-ansible-base/pull/1087)
    

  ### Key design decisions

  | Decision | Rationale |
  |----------|-----------|
  | Serializer-level validation, not model-level | Model validators cannot skip unchanged fields on update (grandfathering). Confirmed by POC
  [AAP-84927](https://redhat.atlassian.net/browse/AAP-84927). |
  | Reject (HTTP 400), not strip | Stripping silently alters user input. Rejection gives clear feedback. |
  | `excluded_fields` for AuthenticatorMap template fields | `organization`, `role`, `team` accept `{% for_attr_value() %}` expansion syntax which matches the
  Tier 2 template injection pattern. |
  | Fix `super().validate()` chains | `AuthenticatorSerializer` and `AuthenticatorMapSerializer` returned `data` without calling `super().validate()`, silently
  bypassing any mixin in the MRO. Safe fix — `ModelSerializer.validate()` is a passthrough. |

[AAP-77215]: https://redhat.atlassian.net/browse/AAP-77215?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## Summary by CodeRabbit

* **New Features**
  * Added enhanced validation for resource names and free-text fields across authentication, OAuth, and role-management workflows.
  * Detects unsafe markup, URI schemes, shell substitutions, template expressions, control characters, and encoded bypasses.
  * Validates text within JSON configuration with exclusions and clear nested error messages.
  * Preserves unchanged legacy values during updates where appropriate.
  * Records validation warnings with relevant request context for auditing.
  * Can be enabled through the enhanced input validation setting.

* **Documentation**
  * Added guidance covering validation behavior, exclusions, JSON fields, auditing, and usage recommendations.

### CODE REVIEW

File: ansible_base/authentication/serializers/authenticator.py [hunk 1]
```diff
@@ -6,12 +6,27 @@
...
+    # JSONField scan.  Other structured sub-keys (DictField, ListField, JSONField) are
+    # already skipped by the mixin's isinstance(val, str) check.
+    # See docs/lib/validation.md "Authenticator configuration exclusions" for the full rationale.
+    excluded_json_keys = {
+        'configuration': frozenset(
+            {
+                'SECRET',
+                'BIND_PASSWORD',
+                'SP_PRIVATE_KEY',
+                'ADDITIONAL_UNVERIFIED_ARGS',
```
Comment (prat98) [inline ansible_base/authentication/serializers/authenticator.py:25]: I believe the mixin would skip `ADDITIONAL_UNVERIFIED_ARGS` anyways as it is a nested JSONField and the mixin only does validation one level deep. We could double check this but I think we'd be covered without adding it here

Comment (vidyanambiar) [reply ansible_base/authentication/serializers/authenticator.py:25]: You're right that the mixin would skip it in the common case. ADDITIONAL_UNVERIFIED_ARGS is typically a dict, and `_validate_json_dict` skips non-strings via `isinstance(val, str)`. The exclusion is intentional for two reasons:

-  ADDITIONAL_UNVERIFIED_ARGS is a JSONField, so it accepts any JSON type. If someone passes a raw string value instead of a dict, the mixin would run validate_free_text() on it without the explicit exclusion. Belt-and-suspenders, but it avoids a surprising rejection on a field that's documented as unvalidated.
-  The field's help text says "not validated and passed directly back to the authenticator." The explicit exclusion makes the code match that contract rather than relying on a type-check side effect.

**Comment by vidyanambiar** [comment]
## Code Review: Add CleanTextMixin to DAB concrete model serializers

**Verdict**: READY_FOR_HUMAN_REVIEW
**Score**: 10/10 (Functionality 10, Security 10, Quality 10)

No critical or major findings. All findings from prior iterations resolved.

### Score Trajectory

```
Run 1: ████████████████████████████████████████████████░░  9.5/10
Run 2: █████████████████████████████████████████████████░  9.7/10
Run 3: ██████████████████████████████████████████████████  10/10
```

### Prior Finding Status
| # | Finding | Status |
|---|---------|--------|
| 1 | Missing `excluded_json_keys` on AuthenticatorSerializer | **FIXED** — added with SECRET, BIND_PASSWORD, SP_PRIVATE_KEY, ADDITIONAL_UNVERIFIED_ARGS |
| 2 | AuthenticatorMap tested at serializer level not API level | **FIXED** — API-level test added |
| 3 | `test_excluded_fields` missing `team` | **FIXED** — `team` added |
| 4 | No comment explaining missing token grandfather test | **FIXED** — comment added |

### Verification Results (19 checks, all PASS)
| Check | Result |
|-------|--------|
| MRO correct — all 5 serializers | PASS |
| super().validate() chains complete | PASS |
| excluded_fields matches _EXPANSION_FIELDS | PASS |
| excluded_json_keys covers all encrypted fields (15 plugins) | PASS |
| ADDITIONAL_UNVERIFIED_ARGS excluded | PASS |
| Non-string config sub-keys auto-skipped | PASS |
| PEM cert fields safe without exclusion | PASS |
| Subclass inheritance (AuthenticatorUpdateSerializer, RoleDefinitionDetailSerializer) | PASS |
| API-level + serializer-level test coverage | PASS |
| ReDoS in regex patterns (tested 100K-char inputs) | PASS |
| Error messages don't reflect input (no XSS) | PASS |
| Audit logging safe (no log injection) | PASS |
| Doc tables match plugin fields | PASS |

### Observations

- The `super().validate()` fix in `AuthenticatorSerializer` and `AuthenticatorMapSerializer` closes a latent bug where any future mixin in the MRO would be silently bypassed.
- The `excluded_json_keys` hardcoded approach (vs. dynamic plugin introspection) avoids import-order complications at class definition time. Maintenance note in `docs/lib/validation.md`.
- The `triggers` JSONField on `AuthenticatorMap` stores nested dicts — automatically skipped by `isinstance(val, str)`.

---
*AI-assisted code review — not a final approval. Human review required.*

**Comment by vidyanambiar** [comment]
Manual Testing Instructions: https://docs.google.com/document/d/19vNXrfC-bnspJKWErvf0jgGStRqTYGmAsu54wWuuXwU/edit?usp=sharing

File: ansible_base/authentication/serializers/authenticator.py (hunk 1 above)
Comment (prat98) [reply ansible_base/authentication/serializers/authenticator.py:25]: That makes sense to me
